barretenberg/cpp/src/barretenberg/vm2/simulation/lib/db_types.hpp

```diff
@@ -3,12 +3,49 @@
 #include "barretenberg/crypto/merkle_tree/indexed_tree/indexed_leaf.hpp"
 #include "barretenberg/crypto/merkle_tree/types.hpp"
 #include "barretenberg/vm2/common/aztec_types.hpp"
+#include "barretenberg/vm2/common/map.hpp"
+#include "barretenberg/vm2/simulation/interfaces/db.hpp"
 
 namespace bb::avm2::simulation {
 
 using NullifierTreeLeafPreimage = crypto::merkle_tree::IndexedLeaf<crypto::merkle_tree::NullifierLeafValue>;
 using PublicDataTreeLeafPreimage = crypto::merkle_tree::IndexedLeaf<crypto::merkle_tree::PublicDataLeafValue>;
 
+// Keys for hints stored in unordered_flat_maps, used in raw_data_dbs and hinting_dbs.
+using GetSiblingPathKey = std::tuple<AppendOnlyTreeSnapshot, MerkleTreeId, index_t>;
+using GetPreviousValueIndexKey = std::tuple<AppendOnlyTreeSnapshot, MerkleTreeId, FF>;
+using GetLeafPreimageKey = std::tuple<AppendOnlyTreeSnapshot, index_t>;
+using GetLeafValueKey = std::tuple<AppendOnlyTreeSnapshot, MerkleTreeId, index_t>;
+using SequentialInsertHintPublicDataTreeKey = std::tuple<AppendOnlyTreeSnapshot, MerkleTreeId, PublicDataLeafValue>;
+using SequentialInsertHintNullifierTreeKey = std::tuple<AppendOnlyTreeSnapshot, MerkleTreeId, NullifierLeafValue>;
+using AppendLeavesHintKey = std::tuple<AppendOnlyTreeSnapshot, MerkleTreeId, std::vector<FF>>;
+
+// TODO(MW): Temp struct for hints to allow using a ref in the HintingContractsDB class constructor
+struct MappedContractHints {
+    unordered_flat_map<AztecAddress, ContractInstanceHint> contract_instances;
+    unordered_flat_map<ContractClassId, ContractClassHint> contract_classes;
+    unordered_flat_map<ContractClassId, BytecodeCommitmentHint> bytecode_commitments;
+};
+
+// TODO(MW): Temp struct for hints to allow using a ref in the HintingRawDB class constructor
+struct MappedMerkleHints {
+    // Query hints:
+    unordered_flat_map<GetSiblingPathKey, GetSiblingPathHint> get_sibling_path_hints;
+    unordered_flat_map<GetPreviousValueIndexKey, GetPreviousValueIndexHint> get_previous_value_index_hints;
+    unordered_flat_map<GetLeafPreimageKey, GetLeafPreimageHint<PublicDataTreeLeafPreimage>>
+        get_leaf_preimage_hints_public_data_tree;
+    unordered_flat_map<GetLeafPreimageKey, GetLeafPreimageHint<NullifierTreeLeafPreimage>>
+        get_leaf_preimage_hints_nullifier_tree;
+    unordered_flat_map<GetLeafValueKey, GetLeafValueHint> get_leaf_value_hints;
+    // State modification hints:
+    unordered_flat_map<SequentialInsertHintPublicDataTreeKey, SequentialInsertHint<PublicDataLeafValue>>
+        sequential_insert_hints_public_data_tree;
+    unordered_flat_map<SequentialInsertHintNullifierTreeKey, SequentialInsertHint<NullifierLeafValue>>
+        sequential_insert_hints_nullifier_tree;
+    unordered_flat_map<AppendLeavesHintKey, AppendLeavesHint> append_leaves_hints;
+    // TODO(MW): Add checkpoint hints here?
```

Comment:
I think yes you'll have to probably? or did you get them to coincide somehow else?